### PR TITLE
feat(metrics): Always sample metrics for slo events

### PR DESCRIPTION
**Pull request by beezz**

### CODE REVIEW

File: src/sentry/tasks/store.py [hunk 1]
```diff
@@ -771,6 +771,35 @@ def _do_save_event(
...
+    ):
+        return
+
+    internals = data["extra"].get("_sentry_internal_metrics")
+    if not internals:
+        return
+
+    metrics.timing(
+        "events.internal-time-to-ingest-total",
+        time() - data["timestamp"],
```
Comment (tonyo) [inline src/sentry/tasks/store.py:790]: Let's do `now = time()` and then reuse it.

File: src/sentry/tasks/store.py [hunk 2]
```diff
@@ -771,6 +771,35 @@ def _do_save_event(
                     "events.time-to-process", time() - start_time, instance=data["platform"]
                 )
 
+            time_internal_metrics_event(data, project_id)
+
+
+def time_internal_metrics_event(data, project_id, start_time):
+    if (
+        settings.SENTRY_INTERNAL_METRICS_PROJECT_ID is None
```
Comment (tonyo) [inline src/sentry/tasks/store.py:779]: To be on the safe side, wdyt about doing `getattr(settings, NAME, None)` here?

File: src/sentry/tasks/store.py [hunk 3]
```diff
@@ -771,6 +771,35 @@ def _do_save_event(
                     "events.time-to-process", time() - start_time, instance=data["platform"]
                 )
 
+            time_internal_metrics_event(data, project_id)
+
+
+def time_internal_metrics_event(data, project_id, start_time):
```
Comment (tonyo) [inline src/sentry/tasks/store.py:777]:
```suggestion
def _time_internal_metrics_event(data, project_id, start_time=None):
```

File: src/sentry/tasks/store.py [hunk 4]
```diff
@@ -771,6 +771,35 @@ def _do_save_event(
                     "events.time-to-process", time() - start_time, instance=data["platform"]
                 )
 
+            time_internal_metrics_event(data, project_id)
```
Comment (tonyo) [inline src/sentry/tasks/store.py:774]:
```suggestion
            _time_internal_metrics_event(data, project_id, start_time)
```

**Review by tonyo — CHANGES_REQUESTED** [review]

File: src/sentry/tasks/store.py [hunk 5]
```diff
@@ -771,6 +771,39 @@ def _do_save_event(
...
+        return
+
+    now = time()
+    tags = {"target": extra["target"], "region": extra["region"], "source": extra["source"]}
+    instance = extra["region"]
+
+    metrics.timing(
+        "events.synthetic-monitoring.time-to-ingest-total",
+        now - data["timestamp"],
+        instance=instance,
```
Comment (tonyo) [inline src/sentry/tasks/store.py:793]: Do we really need this `instance` argument?

File: src/sentry/tasks/store.py [hunk 6]
```diff
@@ -771,6 +771,39 @@ def _do_save_event(
                     "events.time-to-process", time() - start_time, instance=data["platform"]
                 )
 
+            time_synthetic_monitoring_event(data, project_id, start_time)
+
+
+def time_synthetic_monitoring_event(data, project_id, start_time):
```
Comment (tonyo) [inline src/sentry/tasks/store.py:777]: Please let's add a doc string to explain what we do here.

File: src/sentry/conf/server.py [hunk 7]
```diff
@@ -1926,3 +1926,6 @@ def get_sentry_sdk_config():
     "sentry.incidents.endpoints",
 )
 SENTRY_MAIL_ADAPTER_BACKEND = "sentry.mail.adapter.MailAdapter"
+
+# Project ID used by synthetic monitoring
```
Comment (tonyo) [inline src/sentry/conf/server.py:1930]: Let's elaborate a bit here what synthetic monitoring means.

File: src/sentry/tasks/store.py [hunk 8]
```diff
@@ -771,6 +771,39 @@ def _do_save_event(
...
+
+    if start_time:
+        metrics.timing(
+            "events.synthetic-monitoring.time-to-process",
+            now - start_time,
+            instance=instance,
+            tags=tags,
+            sample_rate=1.0,
+        )
+
```
Comment (tonyo) [inline src/sentry/tasks/store.py:806]: WDYT about returning True (or something else) here, to indicate that this function did something? Then we can assert its return value in tests.

**Review by tonyo — CHANGES_REQUESTED** [review]
A few minor comments. Thanks for the tests!

File: src/sentry/tasks/store.py (hunk 5 above)
Comment (beezz) [reply src/sentry/tasks/store.py:793]: I don't think so. Removing, we have enough info in tags.

File: src/sentry/tasks/store.py (hunk 6 above)
Comment (beezz) [reply src/sentry/tasks/store.py:777]: 👍

File: src/sentry/conf/server.py (hunk 7 above)
Comment (beezz) [reply src/sentry/conf/server.py:1930]: 👍

**Review by tonyo — APPROVED** [review]
🏄

**Pull request merged**